### ISSUE TITLE
[Samples] System.TypeInitializationException: The type initializer for `FSharp.Data.GraphQL.Samples.StarWarsApi.Schema` threw an exception

### ISSUE THREAD

**Issue by valbers** [fsprojects/FSharp.Data.GraphQL#376]
### Description

I can't run the Star Wars API sample.

### Repro steps

1. Clone repo to your local machine (macOS, Intel architecture);
2. Switch to branch `dev` (e.g.: `git switch dev`), if you're not already there;
3. Make sure you have a dotnet 5 version installed (e.g.: `brew install dotnet-sdk5-0-400`); 
4. Execute shell script `./build.sh`;
5. `cd` to `samples/star-wars-api`;
6. Execute `dotnet run`

### Expected behavior

Project runs successfully.

### Actual behavior

Program crashes on startup with the following exception:
```
Application startup exception: System.TypeInitializationException: The type initializer for 'FSharp.Data.GraphQL.Samples.StarWarsApi.Schema' threw an exception.
 ---> System.TypeInitializationException: The type initializer for '<StartupCode$FSharp-Data-GraphQL-Samples-StarWarsApi>.$Schema' threw an exception.
 ---> System.InvalidCastException: Unable to cast object of type 'go@77[FSharp.Data.GraphQL.Types.SchemaCompileContext,Microsoft.FSharp.Core.Unit]' to type 'go@77[Microsoft.FSharp.Core.Unit,FSharp.Data.GraphQL.IExecutorMiddleware]'.
   at <StartupCode$FSharp-Data-GraphQL-Server>.$Executor.go@77-1.Invoke(ctx ctx, FSharpList`1 _arg1) in /Users/valber/dev/learning/fsharp/fsharp-data-graphql/src/FSharp.Data.GraphQL.Server/Executor.fs:line 82
   at FSharp.Data.GraphQL.Executor`1.runMiddlewares[ctx,res](FSharpFunc`2 phaseSel, ctx initialCtx, FSharpFunc`2 onComplete) in /Users/valber/dev/learning/fsharp/fsharp-data-graphql/src/FSharp.Data.GraphQL.Server/Executor.fs:line 83
   at FSharp.Data.GraphQL.Executor`1..ctor(ISchema`1 schema, IEnumerable`1 middlewares, FSharpOption`1 validationCache) in /Users/valber/dev/learning/fsharp/fsharp-data-graphql/src/FSharp.Data.GraphQL.Server/Executor.fs:line 87
   at <StartupCode$FSharp-Data-GraphQL-Samples-StarWarsApi>.$Schema..cctor() in /Users/valber/dev/learning/fsharp/fsharp-data-graphql/samples/star-wars-api/Schema.fs:line 236
   --- End of inner exception stack trace ---
   at FSharp.Data.GraphQL.Samples.StarWarsApi.Schema..cctor()
   --- End of inner exception stack trace ---
   at FSharp.Data.GraphQL.Samples.StarWarsApi.Schema.get_executor()
   at FSharp.Data.GraphQL.Samples.StarWarsApi.Startup.Configure(IApplicationBuilder app) in /Users/valber/dev/learning/fsharp/fsharp-data-graphql/samples/star-wars-api/Startup.fs:line 26
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.AspNetCore.Hosting.ConfigureBuilder.Invoke(Object instance, IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.ConfigureBuilder.<>c__DisplayClass4_0.<Build>b__0(IApplicationBuilder builder)
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.Configure(IApplicationBuilder app)
   at Microsoft.AspNetCore.HostFilteringStartupFilter.<>c__DisplayClass0_0.<Configure>b__0(IApplicationBuilder app)
   at Microsoft.AspNetCore.Hosting.WebHost.BuildApplication()
crit: Microsoft.AspNetCore.Hosting.WebHost[6]
```

### Related information 

* Operating system: macOS Monterey (12.5.1)
* Branch: `dev`
* dotnet --info:
```
.NET SDK (gemäß "global.json"):
 Version:   5.0.408
 Commit:    da985e2a23

Laufzeitumgebung:
 OS Name:     Mac OS X
 OS Version:  12.5
 OS Platform: Darwin
 RID:         osx-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.408/

global.json file:
  /Users/valber/dev/learning/fsharp/fsharp-data-graphql/global.json

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

.NET SDKs installed:
  5.0.408 [/usr/local/share/dotnet/sdk]
  6.0.400 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.8 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.17 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.8 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Download .NET:
  https://aka.ms/dotnet-download

Learn about .NET Runtimes and SDKs:
  https://aka.ms/dotnet/runtimes-sdk-info
```

**Comment by xperiandri** [comment]
Interesting. Looks like on that line unit appears instead of schema

**Comment by xperiandri** [comment]
Are you evaluating this library for your project?

**Comment by valbers** [comment]
Yes, I am.

Andrii Chebukin ***@***.***> schrieb am So., 21. Aug. 2022,
12:54:

**Comment by xperiandri** [comment]
The library is working but requires some maintenance. If you are interested to help, we can have a call.
Looks like our issue is pretty simple but we need to have a look

**Comment by valbers** [comment]
Thank you for your solicitude @xperiandri. So far I found out that the offending piece is the middlewares list. If in `samples/star-wars-api/Schema.fs:230` I set the value of variable `middlewares` to an empty (generic) list, then the application starts fine.

For reference, this is how `middlewares` is currently set in the repo (which makes the application throw the exception during initialization):
```
     let middlewares = 
         [ Define.QueryWeightMiddleware(2.0, true)
           Define.ObjectListFilterMiddleware<Human, Character option>(true)
           Define.ObjectListFilterMiddleware<Droid, Character option>(true)
           Define.LiveQueryMiddleware() ]
```

**Comment by xperiandri** [comment]
Thanks, that must be fixed

**Comment by valbers** [comment]
I'm taking a look into it, but unfortunately I can't promise any progress any time soon.
In any case, I'm kind of disappointed to see such an error happening in an F# project. Functional programming should be about "if it compiles, it works". I see, though, that at least some OOP must be made to interface with the ASP.NET infrastructure. But as you can see, the least OOP is enough to create problems already 🙂

**Comment by xperiandri** [comment]
Yes, but at this library level, you are pretty safe 🙂

**Comment by valbers** [comment]
Please review https://github.com/fsprojects/FSharp.Data.GraphQL/pull/377

**Comment by xperiandri** [comment]
Fixed in #377
